gufe/protocols/protocoldag.py

```diff
@@ -85,14 +86,22 @@ class ProtocolDAGResult(GufeTokenizable, DAGMixin):
     _unit_result_mapping: dict[ProtocolUnit, list[ProtocolUnitResult]]
     _result_unit_mapping: dict[ProtocolUnitResult, ProtocolUnit]
 
+    _transformation: Optional[GufeKey]
```

Comment:
I put `_transformation` here as an optional attribute so that it isn't strictly required to run a `ProtocolDAG` and get back a `ProtocolDAGResult`. I still prefer the idea that a `Protocol` developer can write and test their `Protocol` without having to invoke a `Transformation` object at all, but I recognize that this is getting to be a small difference at this point.